### DISCUSSION TITLE
Call WebView in react native

### DISCUSSION THREAD

**Discussion by yuchuga** [jaredpalmer/formik#3603] [Q&A]
Hi, does anyone know how to call "window.ReactNativeWebView.postMessage("Form submitted");" in Formik onSubmit method() such that it will post a message on the mobile screen when calling from WebView after submitting the form. [bombillazo](https://github.com/bombillazo), [anderssonq](https://github.com/anderssonq) can help pls

The link for this https://spin.atomicobject.com/2019/10/28/communicating-react-native-webview/

<img width="711" alt="Screenshot 2022-07-12 at 11 23 37 AM" src="https://user-images.githubusercontent.com/85331903/178401945-bb3035d0-3f34-4b93-8734-ff9def500984.png">

<img width="711" alt="Screenshot 2022-07-12 at 11 29 03 AM" src="https://user-images.githubusercontent.com/85331903/178402508-e8beaf9e-a635-471a-91df-12f307349064.png">

**Comment by bombillazo** [comment]
Hey @yuchuga. Unfortunately, I am no longer using Formik, I migrated our forms to React Hook Form due to certain difficulties with Formik, nor have used WebViews. Try looking to see if web views have some type of listener to attach your function to when a specific event happens. Best of luck!

**Reply by yuchuga** [reply]
Hi @[bombillazo](https://github.com/bombillazo), can react hook forms use WebView ?

**Reply by bombillazo** [reply]
I haven't used WebView but following [this guide](https://www.educative.io/answers/how-to-use-webview-in-react-native), seems it is doable with either Formik or React Hook Form. I'm assuming your form is inside the WebView, so simply call the `window.ReactNativeWebView.postMessage` function inside your WebView JS logic, and pass a function to the `onMessage` prop to handle the data from the WebView.

```jsx
// inside the webview JS logic

<Formik
  // form props
  onsubmit={(values, action, e) => {
    // your form handling logic
    const payload = { id: "test" };
    window.ReactNativeWebView.postMessage(JSON.stringify(payload)); // <--- message NEEDS to be serialized!
  }}
/>

```
```jsx
// inside your React Native logic
<WebView
  source={{uri: 'https://www.example.com'}}
  onMessage={(event) => {
    const serializedPayload = event.nativeEvent.data;
    console.log(JSON.parse(serializedPayload)); // logs the payload object
  }
/>
```

Check the guide and the [WebView API Reference](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Reference.md) on how to inject data to the WebView and more stuff.

**Reply by yuchuga** [reply]
@[bombillazo](https://github.com/bombillazo). Thanks for helping! How do I validate whether the data is being send as a string to the server after submitting the form? Also do you know whether Formik is able to link to AWS amplify and S3 bucket ?

**Reply by yuchuga** [reply]
Hi @[bombillazo](https://github.com/bombillazo), can we arrange a zoom session tmr to discuss about this please. Just ping here when you available. Can I have your email address to send you an invite. Thanks!
